src/common/utils/downloadFile.ts

```diff
@@ -0,0 +1,36 @@
+import request from "request";
+
+export interface DownloadFileOptions {
+  url: string;
+  gzip?: boolean;
+}
+
+export interface DownloadFileTicket {
+  url: string;
+  promise: Promise<Buffer>;
+  cancel(): void;
+}
+
+export function downloadFile(opts: DownloadFileOptions): DownloadFileTicket {
+  const { url, gzip = true } = opts;
+  const fileChunks: Buffer[] = [];
+  const req = request(url, { gzip });
+  const promise: Promise<Buffer> = new Promise((resolve, reject) => {
+    req.on("data", (chunk: Buffer) => {
+      fileChunks.push(chunk);
+    });
+    req.on("complete", () => {
+      resolve(Buffer.concat(fileChunks));
+    });
+    req.on("error", err => {
+      reject({ url, err });
+    });
+  });
+  return {
+    url: url,
```

Comment:
[nit] `url,` is enough - I think we should enable object-shorthand in eslint to make this automatic